packages/tessellate-request/lib/configuration/oauth-configuration-provider.js

```diff
@@ -0,0 +1,16 @@
+// @flow
+
+import type {OAuthConfig} from './oauth-configuration'
+
+import * as oAuthConfigurationLoader from './oauth-configuration-file-loader'
+oAuthConfigurationLoader.init()
+
+export function getById(id: string): OAuthConfig {
+  const config = oAuthConfigurationLoader.get().find((element) => element.id === id)
+
+  if(config === undefined) {
+    throw new Error(`${id} OAuth configuration not found`)
+  }
+
+  return config
+}
```

Comment:
It's good idea to separate these: there may be many loaders (e.g. file://, https://, etc.) but they all return the same JSON structure. Extracting data from the config JSON should be done in a single place (in this case in the`configuration-provider'). If we move the extraction logic to the loader itself, it would need to be duplicated for each loader.